src/gui/qgsexpressionbuilderdialog.cpp

```diff
@@ -83,6 +86,40 @@ void QgsExpressionBuilderDialog::accept()
   QDialog::accept();
 }
 
+void QgsExpressionBuilderDialog::reject()
+{
+
+  if ( builder->expressionText() != mInitialText )
+  {
+
+    QgsSettings settings;
+    const bool askToDiscardEditedExpression = settings.value( QStringLiteral( "askToDiscardEditedExpression" ), true, QgsSettings::Gui ).toBool();
+
+    if ( askToDiscardEditedExpression )
+    {
+      QMessageBox confirmMessage( QMessageBox::Question,
+                                  tr( "Expression was edited" ),
```

Comment:
```suggestion
                                  tr( "Expression was Edited" ),
```

Reply:
Thanks for looking at it :) Why should that word be uppercase?

Reply:
Ah, because of https://docs.qgis.org/testing/en/docs/developers_guide/hig.html :)